core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -216,7 +216,7 @@ public static ProxyConfiguration load() throws IOException {
      */
     public static URLConnection open(URL url) throws IOException {
         Jenkins h = Jenkins.getInstanceOrNull(); // this code might run on slaves
```

Comment:
> `Channel.current` is only valid during serialization / deserialization

Not according to [Javadoc](https://github.com/jenkinsci/remoting/blob/4a6b4ca2f05a99369665a2f9ed391b972f38e95d/src/main/java/hudson/remoting/Channel.java#L1475).